### PR TITLE
[Tidy] Improve semantics of `NavBar` and `Accordion`

### PR DESCRIPTION
**Pull request by huong-li-nguyen**
## Description
- Replace `html.Div` with `dbc.Nav` and `dbc.Navbar` (for context on [why we should do the switch](https://stackoverflow.com/questions/36097957/what-is-the-difference-between-nav-and-div))

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-core/src/vizro/models/_dashboard.py [hunk 1]
```diff
@@ -32,8 +32,13 @@
...
 def _all_hidden(components: List[Component]):
-    """Returns True if all `components` are either None and/or have hidden=True."""
-    return all(component is None or getattr(component, "hidden", False) for component in components)
+    """Returns True if all `components` are either None and/or have hidden=True and/or className contains `d-none`."""
+    return all(
+        component is None
+        or getattr(component, "hidden", False)
+        or "d-none" in getattr(component, "className", "d-inline")
+        for component in components
+    )
```
Comment (huong-li-nguyen) [inline vizro-core/src/vizro/models/_dashboard.py:41]: @antonymilne - I know we've changed this back and forth 😅 I tried to do everything in this commit so it's easier to see which changes are required: https://github.com/mckinsey/vizro/pull/393/commits/74031d621cf2201fda4709f51cfa945e3ba54eab
Essentially the `dbc.Nav` / `dbc.Navbar` do not have a `hidden` argument, so we have to go back to one of the previous approaches of providing a `className` for these. Note I am using the classNames from Bootstrap utilities: https://getbootstrap.com/docs/5.0/utilities/display/. Returning None wouldn't work in this case, as we need to access the the html elements `nav-bar` and ´nav-panel´ via their ID later on.

File: vizro-core/src/vizro/models/_navigation/accordion.py [hunk 2]
```diff
@@ -44,7 +42,7 @@ def build(self, *, active_page_id=None):
         # Note build does not return _NavBuildType but just a single html.Div with id="nav-panel".
         # Hide navigation panel if there is only one page
         if len(list(itertools.chain(*self.pages.values()))) == 1:
-            return html.Div(hidden=True, id="nav-panel")
+            return dbc.Nav(id="nav-panel", className="d-none invisible")
```
Comment (huong-li-nguyen) [inline vizro-core/src/vizro/models/_navigation/accordion.py:45]: `d-none` is equivalent to `display: none`
`invisible` is equivalent to `visibility: hidden`

https://getbootstrap.com/docs/5.0/utilities/display/

File: vizro-core/src/vizro/models/_dashboard.py (hunk 1 above)
Comment (antonymilne) [reply vizro-core/src/vizro/models/_dashboard.py:41]: This looks good, thanks for making the change so clear just in one commit! 🚀 

Just one thing I'm curious about: why do we put `className="d-none invisible"` and not just `d-none`?

**Review by antonymilne — APPROVED** [review]
This is great, thank you!

**Review by maxschulz-COL — APPROVED** [review]
👍

**Pull request merged**